src/formula.rs

```diff
@@ -325,6 +376,48 @@ impl PartialEq for Formula {
     }
 }
 
+/// An iterator through the labels of a formula.
+/// This iterator sequence is depth-first, postfix (post-order).
+/// It provides the label, and a boolean indicating whether the current label is a variable or not.
+#[derive(Debug)]
+pub struct LabelIter<'a> {
+    formula: &'a Formula,
+    stack: Vec<(NodeId, SiblingIter<'a, Label>)>,
+}
+
+impl<'a> Iterator for LabelIter<'a> {
+    type Item = (Label, bool);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut stack_end = self.stack.len() - 1;
+        if self.stack.is_empty() {
+            return None;
+        }
```

Comment:
```suggestion
        if self.stack.is_empty() {
            return None;
        }
        let mut stack_end = self.stack.len() - 1;
```
I think this can also be structured as `while let Some((node_id, iter)) = self.stack.pop() { ... }` with a `push` in the looping case, instead of the index arithmetic

Reply:
I finally changed it to the more natural pre-order, and used a `last_mut()`, which allows to peek the last element of the Vec without popping it.